tests/cstor/volume/waitforfirstconsumer_test.go

```diff
@@ -0,0 +1,203 @@
+				Name:                spcName,
+				DiskType:            "sparse",
+				PoolCount:           cstor.PoolCount,
+				IsThickProvisioning: true,
+				PoolType:            "striped",
+			}
+			ops.Config = spcConfig
+			spcObj = ops.BuildAndCreateSPC()
+			By("Creating SPC, Desired Number of CSP Should Be Created", func() {
+				ops.VerifyDesiredCSPCount(spcObj, cstor.PoolCount)
+			})
+		})
+	})
+
+	When("Apply Volume Related Artifacts", func() {
+		It("Volume Should be Created and Provisioned", func() {
+			By("Build And Create StorageClass")
+
+			casConfig := strings.Replace(
+				openebsCASConfigValue, "$spcName", spcObj.Name, 1)
+			casConfig = strings.Replace(
+				casConfig, "$count", strconv.Itoa(cstor.ReplicaCount), 1)
+			annotations[string(apis.CASTypeKey)] = string(apis.CstorVolume)
+			annotations[string(apis.CASConfigKey)] = casConfig
+			scConfig := &tests.SCConfig{
+				Name:              scName,
+				Annotations:       annotations,
+				Provisioner:       openebsProvisioner,
+				VolumeBindingMode: storagev1.VolumeBindingWaitForFirstConsumer,
+			}
+			ops.Config = scConfig
+			scObj = ops.CreateStorageClass()
+
+			pvcConfig := &tests.PVCConfig{
+				Name:        pvcName,
+				Namespace:   nsObj.Name,
+				SCName:      scObj.Name,
+				Capacity:    "5G",
+				AccessModes: accessModes,
+			}
+			ops.Config = pvcConfig
+			pvcObj = ops.BuildAndCreatePVC()
+		})
+	})
+
+	When("Deploying Percona Application", func() {
+		It("CStor Volume Related Resources Should Be Created and Become Healthy", func() {
+			var err error
+			// Deploying Application
+			createAndDeployAppPod()
+
+			By("Verifying pvc status as bound")
+
+			// Verify health of CStor Volume Related CR's
+			ops.VerifyVolumeStatus(pvcObj,
+				cstor.ReplicaCount,
+				cvr.PredicateList{cvr.IsHealthy()},
+				cv.PredicateList{cv.IsHealthy()},
+			)
+			pvcObj, err = ops.PVCClient.
+				WithNamespace(pvcObj.Namespace).
+				Get(pvcObj.Name, metav1.GetOptions{})
+			Expect(err).To(BeNil())
+		})
+	})
+
+	When("Deleting Application, PVC and SPC", func() {
+		It("Should Delete Volume and Pools Related to test", func() {
+			err := ops.DeployClient.WithNamespace(nsObj.Name).
+				Delete(appDeployment.Name, &metav1.DeleteOptions{})
+			Expect(err).ShouldNot(HaveOccurred(), "while deleting application pod")
+			cnt := ops.GetPodCountEventually(nsObj.Name, "app=busybox", pod.PredicateList{}, 0)
+			Expect(cnt).Should(BeNumerically("==", 0), fmt.Sprintf("While waiting for application pod to scale down"))
+
+			ops.DeletePersistentVolumeClaim(pvcObj.Name, pvcObj.Namespace)
+			ops.VerifyVolumeResources(pvcObj.Spec.VolumeName, openebsNamespace, cvr.PredicateList{}, cv.PredicateList{})
+			err = ops.SCClient.Delete(scObj.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil())
+
+			err = ops.SPCClient.Delete(
+				spcObj.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting spc {%s}", spcObj.Name)
+		})
+	})
+})
+
+func createAndDeployAppPod() {
```

Comment:
can we have this `func CreateDeployment(deployConfig)` in operations.go file as a generic fun the way we did for pvc, SC  etc?